util/test/demos/d3d12/d3d12_descriptor_indexing.cpp

```diff
@@ -255,9 +255,9 @@ float4 main(v2f IN) : SV_Target0
 
     ID3D12ResourcePtr structBuf = MakeBuffer().UAV().Size(8192);
     D3D12_GPU_DESCRIPTOR_HANDLE structGPU =
-        MakeUAV(structBuf).Format(DXGI_FORMAT_R32_UINT).CreateGPU(15);
+        MakeUAV(structBuf).Format(DXGI_FORMAT_R32G32_UINT).CreateGPU(15);
     D3D12_CPU_DESCRIPTOR_HANDLE structCPU =
-        MakeUAV(structBuf).Format(DXGI_FORMAT_R32_UINT).CreateClearCPU(15);
+        MakeUAV(structBuf).Format(DXGI_FORMAT_R32G32_UINT).CreateClearCPU(15);
```

Comment:
Yes this specific format could be left as DXGI_FORMAT_R32_UINT since CreateClearCPU will automatically adjust the number of elements to be doubled ( at half the real size ).

However the previous line : 
> MakeUAV(structBuf).Format(DXGI_FORMAT_R32G32_UINT).CreateGPU(15);

need to be changed because it is used by ClearUnorderedAccessViewUint and for writing/reading the tex_ref between the compute shader and pixel shader.

I mainly changed both to keep them coherent, I don't think ClearUnorderedAccessViewUint impose both descriptors to be identical.

Reply:
Ahh I see what you mean. Then actually a different fix is needed - it's not valid to use a typed buffer (like R32G32_UINT) with a structured buffer binding. We need one descriptor that's typed for clearing (since you can't clear with a structured descriptor), which can still just be R32_UINT, and a different descriptor that's structured with `.Structured(8)` like the SRV.